flowzone.yml

```diff
@@ -1909,112 +1887,112 @@ jobs:
-          fi
-          echo "access=${access}" >> "${GITHUB_OUTPUT}"
+            core.setOutput('private', json.private);
+            core.setOutput('docs', json.scripts.doc);
+
+            if (context.repo.private) {
+              core.setOutput('access', 'restricted');
+            } else {
+              core.setOutput('access', 'public');
+            }
+
+            return json;
 
       # check which past and current and future Node.js LTS releases meet the engine requirements
       # if there are no engine requirements then the current LTS will be used
 
       - <<: *setupNode
-        if: steps.npm.outputs.enabled == 'true'
         with:
           node-version: 18.x
 
-      - name: Check engine
-        if: steps.npm.outputs.enabled == 'true'
+      # https://www.npmjs.com/package/check-engine
+      - &checkEngine
+        name: Check engine
+        id: check_engine_18
         run: |
+          node_version="$(node --version)"
           if npx -q -y -- check-engine
           then
-            echo "NODE_VERSIONS=$(echo "${NODE_VERSIONS}" | jq -c '. + ["18.x"]')" >> "${GITHUB_ENV}"
+            echo "node_version=${node_version}" >> "${GITHUB_OUTPUT}"
           fi
 
       - <<: *setupNode
-        if: steps.npm.outputs.enabled == 'true'
         with:
           node-version: 20.x
 
-      - name: Check engine
-        if: steps.npm.outputs.enabled == 'true'
-        run: |
-          if npx -q -y -- check-engine
-          then
-            echo "NODE_VERSIONS=$(echo "${NODE_VERSIONS}" | jq -c '. + ["20.x"]')" >> "${GITHUB_ENV}"
-          fi
+      - <<: *checkEngine
+        id: check_engine_20
 
       - <<: *setupNode
-        if: steps.npm.outputs.enabled == 'true'
         with:
           node-version: 22.x
 
-      - name: Check engine
-        if: steps.npm.outputs.enabled == 'true'
-        run: |
-          if npx -q -y -- check-engine
-          then
-            echo "NODE_VERSIONS=$(echo "${NODE_VERSIONS}" | jq -c '. + ["22.x"]')" >> "${GITHUB_ENV}"
-          fi
+      - <<: *checkEngine
+        id: check_engine_22
 
-      # default to the current LTS version if none were matched
-      # by the engine checks above
+      # Default to 20.x if no versions were matched
+      # https://github.com/actions/github-script
       - name: Set Node.js versions
-        if: steps.npm.outputs.enabled == 'true'
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7
         id: node_versions
-        run: |
-          echo "json=[\"20.x\"]" >> "${GITHUB_OUTPUT}"
-          if [ "${NODE_VERSIONS}" != "[]" ]
-          then
-            echo "json=${NODE_VERSIONS}" >> "${GITHUB_OUTPUT}"
-          fi
+        env:
+          NODE_VERSIONS: ${{ toJSON(steps.*.outputs.node_version) }}
+        with:
+          result-encoding: json
+          script: |
+            const { compareVersions } = require('util');
+            const versions = JSON.parse(process.env.NODE_VERSIONS) || '["20.x"]';
```

Comment:
This will throw if `NODE_VERSIONS === ''`, I don't know if that's a problem/possible though

Reply:
I was meaning to test that, I'll do so know and handle the throw better. Good catch!

Reply:
Tested [here](https://github.com/product-os/flowzone/actions/runs/12695225997/job/35386913030?pr=1315).

It did not throw, because NODE_VERSIONS was `[]`, but it also failed to fall back to `["20.x"]` so I've pushed a change to re-test.